### PR TITLE
Add circle.yml and static analysis (detekt and lint)

### PR DESCRIPTION
**Pull request by PetarMarijanovic**

### CODE REVIEW

File: config/lint/lint-config.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+<lint>
+
+    <issue id="all">
+        <ignore path="build" />
+    </issue>
+
+    <!-- Remove this 'UnusedResources' when Lint for Kotlin is updated
+    Android Lint: False-positive Lint warning "unused resources" for resources in Kotlin code
+    https://youtrack.jetbrains.com/issue/KT-12499 -->
+    <issue id="UnusedResources" severity="ignore" />
```
Comment (ivanTrogrlic) [inline config/lint/lint-config.xml:11]: What is this for, is it needed?

File: config/lint/lint.gradle [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    lintOptions {
+        lintConfig file("$project.rootDir/config/lint/lint-config.xml")
+
+        warningsAsErrors true
+        abortOnError true
+
+        xmlReport false
+        htmlOutput file("$project.buildDir/reports/lint/lint.html")
+
+        // disable lint because it does not work well with java8 lambda functions
```
Comment (ivanTrogrlic) [inline config/lint/lint.gradle:11]: Im not using StreamSupport. Is this needed?

File: .idea/modules.xml [hunk 3]
```diff
@@ -3,8 +3,11 @@
   <component name="ProjectModuleManager">
     <modules>
       <module fileurl="file://$PROJECT_DIR$/HintedTextView.iml" filepath="$PROJECT_DIR$/HintedTextView.iml" />
+      <module fileurl="file://$PROJECT_DIR$/HintedTextView.iml" filepath="$PROJECT_DIR$/HintedTextView.iml" />
```
Comment (ivanTrogrlic) [inline .idea/modules.xml:6]: This line already exist, why was it added?

**Review by ivanTrogrlic — COMMENTED** [review]
I got a few questions

File: .idea/modules.xml (hunk 3 above)
Comment (ivanTrogrlic) [reply .idea/modules.xml:6]: Actually, this file should be ignored since its in `.idea` folder.

File: config/lint/lint-config.xml (hunk 1 above)
Comment (PetarMarijanovic) [reply config/lint/lint-config.xml:11]: Yes. Lint doesn't work well with Kotlin yet. It reports all strings and drawables as unused even though they are referenced in .kt files.

File: config/lint/lint.gradle (hunk 2 above)
Comment (PetarMarijanovic) [reply config/lint/lint.gradle:11]: Unfortunately, I don't know :S This is the Lint file from our project. We can maybe delete it. Try it out and then let me now :)

File: .idea/modules.xml (hunk 3 above)
Comment (PetarMarijanovic) [reply .idea/modules.xml:6]: Yes, I added `.idea/` to `.gitignore` but it seems that I accidentally added it to the commit before :) You can ignore this or delete if you want to

**Review by ivanTrogrlic — APPROVED** [review]
Thanks for setting it up :)

**Pull request merged**